### ISSUE TITLE
`Style/ArrayIntersect` should also detect `.intersection`

### ISSUE THREAD

**Issue by tejasbubane** [rubocop/rubocop#13986]
## Is your feature request related to a problem? Please describe.

`Style/ArrayInterection` does not detect use of `.intersection`:

```ruby
foo.intersection(bar).any?
```

## Describe the solution you'd like

Cop should detect above case.

## Additional Context

```
rubocop -V
1.73.2 (using Parser 3.3.7.1, rubocop-ast 1.38.1, analyzing as Ruby 3.2, running on ruby 3.4.2) [arm64-darwin24]
```
